spring-rabbit/src/main/java/org/springframework/amqp/rabbit/support/PublisherCallbackChannelImpl.java

```diff
@@ -691,32 +687,33 @@ private synchronized void generateNacksForPendingAcks(String cause) {
 				logger.debug("Added listener " + listener);
```

Comment:
The code not shown here on GH because of without changes, but:
```
if (!this.listeners.values().contains(listener)){
	this.listeners.put(listener.getUUID(), listener);
```
So, listener can change its `UUID` at runtime?
Why `puIfAbsent` isn't good here?
At least by the key hash nature we get a better performance.